test/core/security/evaluate_args_test.cc

```diff
@@ -68,6 +68,204 @@ TEST_F(EvaluateArgsTest, TestEvaluateArgsPeerPort) {
+  grpc_metadata_batch_destroy(&metadata);
+  EXPECT_EQ(evalArgs.GetPath(), "/foo/bar")
+      << "Failed to properly set or retrieve path.";
+  grpc_shutdown();
+}
+
+TEST(EvaluateArgsMetadataTest, GetHostSuccess) {
+  grpc_metadata_batch metadata;
+  grpc_metadata_batch_init(&metadata);
+  grpc_slice fake_val = grpc_slice_intern(grpc_slice_from_static_string("foo"));
+  grpc_mdelem fake_val_md = grpc_mdelem_from_slices(GRPC_MDSTR_HOST, fake_val);
+  grpc_linked_mdelem storage;
+  storage.md = fake_val_md;
+  ASSERT_EQ(grpc_metadata_batch_link_head(&metadata, &storage), nullptr);
+  EvaluateArgs evalArgs(&metadata, nullptr, nullptr);
+  EXPECT_EQ(evalArgs.GetHost(), "foo")
+      << "Failed to properly set or retrieve host.";
+  grpc_metadata_batch_destroy(&metadata);
+}
+
+TEST(EvaluateArgsMetadataTest, GetMethodSuccess) {
+  grpc_metadata_batch metadata;
+  grpc_metadata_batch_init(&metadata);
+  grpc_slice fake_val = grpc_slice_intern(grpc_slice_from_static_string("GET"));
+  grpc_mdelem fake_val_md =
+      grpc_mdelem_from_slices(GRPC_MDSTR_METHOD, fake_val);
+  grpc_linked_mdelem storage;
+  storage.md = fake_val_md;
+  ASSERT_EQ(grpc_metadata_batch_link_head(&metadata, &storage), nullptr);
+  EvaluateArgs evalArgs(&metadata, nullptr, nullptr);
+  EXPECT_EQ(evalArgs.GetMethod(), "GET")
+      << "Failed to properly set or retrieve method.";
+  grpc_metadata_batch_destroy(&metadata);
+}
+
+TEST(EvaluateArgsMetadataTest, GetHeadersSuccess) {
+  grpc_metadata_batch metadata;
+  grpc_metadata_batch_init(&metadata);
+  grpc_slice fake_path =
+      grpc_slice_intern(grpc_slice_from_static_string("/foo/bar"));
+  grpc_mdelem fake_path_md =
+      grpc_mdelem_from_slices(GRPC_MDSTR_PATH, fake_path);
+  grpc_linked_mdelem storage;
+  storage.md = fake_path_md;
+  ASSERT_EQ(grpc_metadata_batch_link_head(&metadata, &storage, GRPC_BATCH_PATH),
+            nullptr)
+      << "couldn't add metadata";
+
+  grpc_slice fake_host =
+      grpc_slice_intern(grpc_slice_from_static_string("foo"));
+  grpc_mdelem fake_host_md =
+      grpc_mdelem_from_slices(GRPC_MDSTR_HOST, fake_host);
+  grpc_linked_mdelem storage2;
+  storage2.md = fake_host_md;
+  ASSERT_EQ(
+      grpc_metadata_batch_link_tail(&metadata, &storage2, GRPC_BATCH_HOST),
+      nullptr)
+      << "couldn't add metadata";
+
+  grpc_slice fake_method =
+      grpc_slice_intern(grpc_slice_from_static_string("GET"));
+  grpc_mdelem fake_method_md =
+      grpc_mdelem_from_slices(GRPC_MDSTR_METHOD, fake_method);
+  grpc_linked_mdelem storage3;
+  storage3.md = fake_method_md;
+  ASSERT_EQ(
+      grpc_metadata_batch_link_head(&metadata, &storage3, GRPC_BATCH_METHOD),
+      nullptr)
+      << "couldn't add metadata";
+
+  EvaluateArgs evalArgs(&metadata, nullptr, nullptr);
+  std::multimap<absl::string_view, absl::string_view> headers =
+      evalArgs.GetHeaders();
+  ASSERT_TRUE(headers.size() == 3) << "number of metdata elements is incorrect";
```

Comment:
The headers test can add just two elements instead of three? The extra attribute doesn't add much value.